### ISSUE TITLE
nvidia-docker2?

### ISSUE THREAD

**Issue by wedgef5** [plumbee/nvidia-virtualgl#3]
Has anyone gotten nvidia-docker2 and VirtualGL working together?

**Comment by pigiuz** [comment]
Hi @wedgef5 
I haven't tried this with nvidia-docker 2.0 yet.
Did you encounter any problem using it there?

**Comment by wedgef5** [comment]
Yes, I'm unable to to get graphical applications to run with hardware acceleration.  First, it appears that nvidia-docker2 is installed and working properly, as the `nvidia-smi` test from the nvidia-docker project correctly detects my video card.  However, when I try to run `vglrun glxinfo`, I get this...

name of display: :0
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
[VGL] WARNING: The OpenGL rendering context obtained on X display
[VGL]    :0 is indirect, which may cause performance to suffer.
[VGL]    If :0 is a local X display, then the framebuffer device
[VGL]    permissions may be set incorrectly.

`vglrun glxgears` also fails to run, giving the same messages as `glxinfo`, plus this...

Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
[VGL] ERROR: in readPixels--
[VGL]    398: GL_ARB_pixel_buffer_object extension not available

I can run just `glxgears` from the container, but it is obviously not accelerated.

Thanks for any tips.

**Comment by hzyjerry** [comment]
I don't think nvidia-docker2 current support OpenGL/VirtualGL
See: [https://github.com/NVIDIA/nvidia-docker/issues/534](url)

You need to uninstall nvidia-docker2 and use v1.0.1 instead:
`wget -P /tmp https://github.com/NVIDIA/nvidia-docker/releases/download/v1.0.1/nvidia-docker_1.0.1-1_amd64.deb
sudo dpkg -i /tmp/nvidia-docker*.deb && rm /tmp/nvidia-docker*.deb`

It works for me

**Comment by romangrothausmann** [comment]
With all the info from various threads I'm confused about the current status: Is nvidia-docker2 still a problem when using e.g. nvidia/opengl:1.0-glvnd-devel-ubuntu16.04 in conjunction with VirtualGL?
https://github.com/NVIDIA/nvidia-docker/issues/534 has evolved quit a bit since @hzyjerry referenced it.
Currently I am able to get openGL to work on a machine with direct X and nvidia-docker2 but not on a machine where VirtualGL is used for rendering. Is GPU isolation required, i.e. driver 390+?
